src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -902,6 +842,30 @@ inline void write_output_segments(Segments& output_segments, const std::shared_p
         output_segments.push_back(std::make_shared<ReferenceSegment>(dummy_table, column_id, pos_list));
       }
     } else {
+      // Check if the PosList references a single chunk. This is easier than tracking the flag through materialization,
+      // radix partitioning, and so on. Also, actually checking for this property instead of simply forwarding it may
+      // allows us to set guarantee_single_chunk in more cases. In cases where more than one chunk is referenced, this
+      // should be cheap. In the other cases, the cost of iterating through the PosList are likely to be amortized in
+      // following operators.
+      // This is not part of PosList as other operators should have a better understanding of how they emit references.
```

Comment:
I can't follow. Who says that the SMJ cannot do this optimization, too?

Reply:
It can, but probably nobody will do that in the near future. That's just a side note from my side. It should never hinder us to optimize the hash join.
But I am still sad that over time the SMJ will be less of an alternative. SAD.

Reply:
Feel free to resolve.